Tests/CommandsTests/PackageCommandTests.swift

```diff
@@ -1067,6 +1067,65 @@ final class PackageCommandTests: CommandsTestCase {
             XCTAssertMatch(contents, .contains(#""OtherLib""#))
         }
     }
+    
+    func testPackageAddRemoteBinaryTarget() async throws {
+        try await testWithTemporaryDirectory { tmpPath in
+            let fs = localFileSystem
+            let path = tmpPath.appending("PackageB")
+            try fs.createDirectory(path)
+
+            try fs.writeFileContents(path.appending("Package.swift"), string:
+                """
+                // swift-tools-version: 5.9
+                import PackageDescription
+                let package = Package(
+                    name: "client"
+                )
+                """
+            )
+
+            _ = try await execute(["add-target", "client", "--url", "https://example.com/client.xcframework.zip", "--checksum", "abcdef1234567890", "--type", "binary"], packagePath: path)
```

Comment:
Would it make sense to add checksum validation to ensure that the user is inputting a valid SHA-256 checksum? As called out in `PackageDescription.Target.checksum`